src/pages/Schedule.tsx

```diff
@@ -1,7 +1,320 @@
-import React from "react";
+import {
+  Box,
+  Button,
+  Container,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Heading,
+  Input,
+  Radio,
+  RadioGroup,
+  Select,
+  Slider,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+  Stack,
+  Textarea,
+  Tooltip,
+  Table,
+  Thead,
+  Tbody,
+  Tfoot,
+  Tr,
+  Th,
+  Td,
+  TableCaption,
+  TableContainer,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Link } from "react-router-dom";
+import { gql, useMutation } from "@apollo/client";
+import { useLoginStore } from "../stores/login";
+
+// These schemas will probably change later, all just example data
+const SUBMIT = gql`
+  mutation Login($username: STring!, $password: STring!) {
+    login(username: $username, password: $password) {
+      username
+      email
+      roles
+    }
+  }
+`;
 
 export const Schedule = () => {
+  const [selectValue, setSelectValue] = useState("");
+  const [submit, { data, loading, error }] = useMutation(SUBMIT);
+
+  const loginState = useLoginStore();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (loginState.loggedIn) {
+      navigate("/");
+    }
+  }, [loginState.loggedIn, navigate]);
+
+  useEffect(() => {
+    if (data && !error && !loading) {
+      // TODO: This route will change to whatever The default page is once The
+      // user is logged in
+      navigate("/");
+    }
+  }, [data, error, loading, navigate]);
+
   return (
-    <></>
+    <Flex
+      w="100%"
+      minH="calc(100vh - 5.5rem)"
+      pt={30}
+      alignItems="center"
+      justifyContent="center"
+      flexDirection="column"
+    >
+      <Container mb={32} maxW="container.xlg">
+        <Heading mb={6}>View Schedule</Heading>
+        <Button
+        mt={5}
+        w="300px"
+        as={Link}
+        to="/schedule"
+        
+        backgroundColor="#8e44ad"
+        colorScheme="purple"
+        textColor="#ffffff"
+        variant="solid"
+        float={`right`}
+      >
+        Generate New Schedule
+      </Button>
+        <Select
+          id="select"
+          w="160px"
+          value={selectValue}
+          mb={5}
+          onChange={(e) => setSelectValue(e.target.value)}
+        >
+          <option value="table">Table View</option>
+          <option value="calendar">Calendar View</option>
+        </Select>
+        <Flex
+          bg="#212938"
+          p={10}
+          borderRadius={10}
+          flexDir="column"
+          style={{ boxShadow: "0px 0px 30px rgba(0, 0, 0, 0.40)" }}
+        >
+         <TableContainer
+          overflowY="auto"
+         >
+            <Table 
+            size='sm'
+            variant="striped"
+            colorScheme="gray"
+            >
```

Comment:
Indentation is weird here, are you using Prettier?

Reply:
yeah i have it installed, do i have to activate it though?

Reply:
I actually don't know because I don't use VSCode lol, but my understanding was that with VSCode plugins you are supposed to be able to install it and it "just works".

In my editor I just have it configured to run prettier with all default settings every time I save the file so it formats automatically. That tends to be the default behavior so I'm surprised it doesn't work like that in VSCode. I think @Adamk8 and @cbuchko both have it configured so maybe you could ask them?